pkg/gofr/testutil/port.go

```diff
@@ -21,3 +23,37 @@ func GetFreePort(t *testing.T) int {
 
 	return port
 }
+
+// ServiceConfigs holds the configuration details for different server components.
+type ServiceConfigs struct {
+	HTTPPort    int
+	HTTPHost    string
+	MetricsPort int
+	MetricsHost string
+	GRPCPort    int
+	GRPCHost    string
+}
+
+// ServerConfigsProvider sets up server configurations for testing and returns a ServiceConfigs struct.
+// It dynamically assigns free ports for HTTP, Metrics, and gRPC services, sets up environment variables for them,
+// and returns a struct with the configured values.
+func ServerConfigsProvider(t *testing.T) *ServiceConfigs {
```

Comment:
I'm unsure about this name

Wording is a pain.

Maybe be this?

```suggestion
func NewServerConfigs(t *testing.T) *ServiceConfigs {
```